functions.js

```diff
@@ -0,0 +1,106 @@
+const {dbConnection} = require("./db");
+
+const getAllStations = async (req, res) => {
+    const db = await dbConnection
+    try {
+        const stations = await db.query('SELECT DISTINCT `code`, `name` FROM stations Order By `name` ASC')
+        res.status(200).json({message: "Successfully Retrieved Stations.", data: stations})
+    } catch (e) {
+        res.status(500).json({message: "Unexpected Error", data: []})
+    }
+}
+
+const getJourneys = async (req, res) => {
+    try {
+        // Get Query Params & Check they're populated. Throw Bad Request.
+        const {origin: originSelection, destination: destinationSelection} = req.query
+        if (!originSelection || !destinationSelection) {
+            res.status(400).json({message: "Origin and Destination are required."})
+        } else {
+            // Unique Stations on Dropdown requires selection of all instances of Origins and Destinations...
+            const origins = await getStationInstances(originSelection)
+            const destinations = await getStationInstances(destinationSelection)
+
+            // Extract Lines to See if Single Line is Possible
+            const originLines = origins.map(station => station.line)
+            const destinationLines = destinations.map(station => station.line)
+
+            // Create Pairs of Origins and Destinations. Stations Exist across multiple Lines.
+            const pairs = originLines.reduce((accumulator, line, originIndex) => {
+                const destinationIndex = destinationLines.indexOf(line)
+                if (destinationIndex !== -1) {
+                    accumulator.push([originIndex, destinationIndex])
+                }
+                return accumulator
+            }, [])
+
+            // Create Routes populated with Origin and Destination station Details
+            let routes = pairs.map(([originIndex, destinationIndex]) => {
+                const originStation = origins[originIndex];
+                const destinationStation = destinations[destinationIndex];
+
+                return {
+                    line: originStation.line,
+                    start: Math.min(originStation.position, destinationStation.position),
+                    end: Math.max(originStation.position, destinationStation.position)
+                }
+            })
+
+            //Query Journey Data
+            const journeys = []
+            for (const route of routes) {
+                let journey = await getSingleLineJourney(route.line, route.start, route.end);
+                journeys.push(journey);
+            }
+
+            //Summarise data to provide information in the format of end Route Card
+            const journeySummaries = []
+
+            for (const journey of journeys) {
+
+                let journeyArrayLength = journey.length - 1
```

Comment:
Rather than doing this whole thing, better to just do these assignments to reuse:

`const fromStation = journey[0]`
`const toStation = journey[journey.length - 1]`